src/lib/support/logging/Constants.h

```diff
@@ -2,6 +2,22 @@
 
 #pragma once
 
+/**
+ * gcc and clang provide a way to warn for a custom formatter when formats don't
+ * match arguments.  Use that for Log() so we catch mistakes.  The "format"
+ * attribute takes the type of format, which arg is the format string, and which
+ * arg is the first variadic arg, with both arg numbers 1-based.
+ *
+ * The second arg should be set to 0 if the function takes a va_list instead of
+ * varargs.
+ */
+
+#if defined(__GNUC__)
+#define ENFORCE_FORMAT(n, m) __attribute__((format(printf, n, m)))
```

Comment:
nit: these do not seem to be "logging constants". Maybe a separate header like `support/enforce_format.h` or `core/printf_format.h` would be clearer.

Reply:
Fair point.  Will move to `support/EnforceFormat.h`.